open-craft/openedx-completion-aggregator
# Aggregate Completion Views

This is moved over from edx/edx-platform#17015
# OC-3096  Aggregate Completion Views (WIP)

**Description:**  Moved PR to new repo.  

This still has some hard links to edx-platform, which are keeping the tests from running.

**JIRA:** 
*   OC-3096 - Completion views 
*   OC-3571 - Move completion aggregator to separate repo.

## Original Description

Code is adapted from the solutions branch v0 api.  Currently has detail and list views working, but not block update view.

Some issues still exist around handling missing completion records, but those won't be resolved until the actual calculation of AggregateCompletion blocks is implemented (see comments on expectedFailure tickets).

The branch is based on cornwell/create-model  (#16715).  Once that merges, this will be re-pointed to edx:master.

### Test directions

- Manually create AggregateCompletion objects with aggregation_name `'course'`, and others with the same course_key and user, with an aggregation_name of `'chapter'`, `'sequential'`, or `'vertical'`.
- Visit /api/completion/v1/course/{course_key}/, and verify that the course record shows up.
- Add the querystring ?requested_fields=chapter,sequential,vertical (or some combination of those), and verify that the other AggregateCompletion records show up.
- Create some AggregateCompletion records in another course.
- Visit /api/completion/v1/course/?requested_fields=chapter,sequential,vertical and verify that the aggregate completions for both courses show up in the appropriate structure.

### Reviewers


FYI: @bradenmacdonald
